### PR TITLE
bump-formula-pr: return remote url with token in it

### PR DESCRIPTION
**Pull request by dawidd6**
Might be useful to be able to push to remote using only the token.
For example on CI.

- [ ] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
> Might be useful to be able to push to remote using only the token.
> For example on CI.

Can you provide a bit more of an example of this? Thanks!

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 1]
```diff
@@ -420,7 +420,12 @@ def forked_repo_info(formula, tap_full_name, backup_file)
     remote_url = if system("git", "config", "--local", "--get-regexp", "remote\..*\.url", "git@github.com:.*")
       response.fetch("ssh_url")
     else
-      response.fetch("clone_url")
+      url = response.fetch("clone_url")
+      if GitHub.env_token
+        url.gsub("https://github.com", "https://#{GitHub.env_token}@github.com")
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb]:
```suggestion
        url.gsub(%r{^https://github\.com/}", "https://#{GitHub.env_token}@github.com/")
```

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 2]
```diff
@@ -420,7 +420,12 @@ def forked_repo_info(formula, tap_full_name, backup_file)
...
+      else
+        url
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb]:
```suggestion
      url.gsub(%r{^https://github\.com/}, "https://#{GitHub.env_token}@github.com/") if GitHub.env_token
      url
```

**Comment by dawidd6** [comment]
On CI one would not need to set up a ssh key to be able to push the branch, but simply just use the token for that, which already has the required scopes.

I'm currently building a Github Action for `brew bump-formula-pr` and this approach seems to be the best option and it works well.

**Comment by MikeMcQuaid** [comment]
> On CI one would not need to set up a ssh key to be able to push the branch, but simply just use the token for that, which already has the required scopes.

Cool, that sounds good then 👍

**Comment by MikeMcQuaid** [comment]
Thanks again @dawidd6!

**Pull request merged**